.genx/templates/main/main.hbs

```diff
@@ -48,8 +52,12 @@ export class MainApplication extends EventEmitter<StoreEventDetailMap>(FASTEleme
       configureDesignSystem(this.provider, designTokens);
     });
 
-    {{#if fdc3}}
-    console.log('FDC3 listening logic here');
+    {{#if data.fdc3ListenChannel}}
+        if (window.fdc3) {
+            await this.fdcReadyHandler();
+        } else {
+            window.addEventListener('fdc3Ready', async () => this.fdcReadyHandler());
+        }
```

Comment:
We could extract this into another little helper in fdc3 utils:

```
{{#if data.fdc3ListenChannel}}
onFDC3Ready(this.fdcReadyHandler);
{{/if}}
```

Reply:
listen to channel is already abstracted - you might need access to `this` or something in the class.

Reply:
I meant subscribing to fdc3 ready event - there's some boilerplate we can do away with. You'll still have access to this if needed.

Reply:
Yes, but need to pass the function like `() => this.fdcReadyHandler()`